core/src/main/java/org/apache/calcite/sql/SqlTableRef.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ *  A <code>SqlTableRef</code> is a node of a parse tree which represents
+ *  a table reference.
+ *
+ *  <p>It can be attached with a sql hint statement, see {@link SqlHint} for details.
+ */
+public class SqlTableRef extends SqlCall {
```

Comment:
`tableRef` can be seen as a table reference but not related to the `hints`, use a more explicitly one: `SqlTableWithHint` ?